src/StringCompression.h

```diff
@@ -32,10 +35,13 @@ namespace Compression
 		/** Returns a view (of type std::byte) of the internal store. */
 		ContiguousByteBufferView GetView() const;
 
-		/** A store allocated on either the stack or heap. */
-		std::variant<Static, Dynamic> Storage;
+		/** A store allocated on the heap.
+		This uses a smart pointer. */
+		std::unique_ptr <std::vector <std::byte>> Storage;
```

Comment:
`std::unique_ptr <std::vector <std::byte>> Storage;`
As suggested, the data type now uses a `std::vector<std::byte>`.
To allocate on the heap, this has been declared using a smart pointer, specifically `std::unique_ptr`